test/distributed/test_c10d_nccl.py

```diff
@@ -575,6 +575,38 @@ def test_set_nccl_pg_timeout(self, backend):
         c10d.distributed_c10d._set_pg_timeout(timedelta(seconds=252), pg)
         self._check_nccl_timeout(timedelta(seconds=252))
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    @parametrize("backend", [None, "nccl"])
+    def test_extend_nccl_pg_timeout(self, backend):
+        torch.cuda.set_device(self.rank)
+        store = c10d.FileStore(self.file_name, self.world_size)
+        opts = dict(
+            backend=backend,
+            store=store,
+            rank=self.rank,
+            world_size=self.world_size,
+            timeout=timedelta(seconds=123),
+        )
+        dist.init_process_group(**opts)
+        pg = dist.distributed_c10d._get_default_group()
+        pg.allreduce(torch.rand(10).cuda(self.rank))
+        pg._get_backend(torch.device(f"cuda:{self.rank}"))._set_default_timeout(
+            timedelta(seconds=3)
+        )
+        if self.rank == 0:
+            # Ideally we want to sleep for a very long time, but this is not
+            # feasible in unit test. So this is only a very tiny case.
+            time.sleep(6)
+            pg.allreduce(torch.rand(10).cuda(self.rank))
+            self._check_nccl_timeout(timedelta(seconds=3))
+        else:
+            dist.distributed_c10d._extend_timeout_all_pgs(timedelta(seconds=100))
+            self._check_nccl_timeout(timedelta(seconds=103))
+            pg.allreduce(torch.rand(10).cuda(self.rank))
+            dist.distributed_c10d._extend_timeout_all_pgs(timedelta(seconds=-10000))
+            self._check_nccl_timeout(timedelta(seconds=600))
```

Comment:
I'm not sure it's very intuitive to do 103 - 10000 and get 600 as a result. What's going on here?

Reply:
Well, it's for users to reset the timeout and the default one for NCCL is 600, as mentioned in the RFC doc.